src/orders/controller/order.controller.ts

```diff
@@ -0,0 +1,146 @@
+import { ApiOperation, ApiParam, ApiResponse, ApiTags } from '@nestjs/swagger';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  Param,
+  Patch,
+  Post,
+} from '@nestjs/common';
+import { CreateOrderDto } from '../dto/order/create-order.dto';
+import { OrderMapper } from '../mapper/order.mapper';
+import { OrderService } from '../service/order.service';
+import { UpdateOrderDto } from '../dto/order/update-order.dto';
+import { OrderDto } from '../dto/order/order.dto';
+
+@ApiTags('orders')
+@Controller('orders')
+export class OrderController {
+  constructor(private readonly orderService: OrderService) {}
+
+  @Get(':orderId')
+  @HttpCode(200)
+  @ApiOperation({ summary: 'Retrieve an order by ID' })
+  @ApiParam({
+    name: 'orderId',
+    description: 'The unique identifier of the order',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'Order retrieved successfully',
+    type: OrderDto,
+  })
+  @ApiResponse({ status: 404, description: 'Order not found' })
+  async getOrderById(@Param('orderId') orderId: string): Promise<OrderDto> {
+    const order = await this.orderService.getById(orderId);
+    return OrderMapper.toDto(order);
+  }
+
+  @Get('customer/:customerId')
+  @HttpCode(200)
+  @ApiOperation({ summary: 'Retrieve orders by customer ID' })
+  @ApiParam({
+    name: 'customerId',
+    description: 'The unique identifier of the customer',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'Orders retrieved successfully',
+    type: [OrderDto],
+  })
+  @ApiResponse({ status: 404, description: 'Customer not found' })
+  async getOrdersByCustomerId(
+    @Param('customerId') customerId: string,
+  ): Promise<OrderDto[]> {
+    const orders = await this.orderService.getOrdersByCustomerId(customerId);
+    return orders.map((order) => OrderMapper.toDto(order));
+  }
+
+  @Get('product/:productId')
+  @HttpCode(200)
+  @ApiOperation({ summary: 'Retrieve orders by product ID' })
+  @ApiParam({
+    name: 'productId',
+    description: 'The unique identifier of the product',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'Orders retrieved successfully',
+    type: [OrderDto],
+  })
+  @ApiResponse({ status: 404, description: 'Product not found' })
+  async getOrdersByProductId(
+    @Param('productId') productId: string,
+  ): Promise<OrderDto[]> {
+    const orders =
+      await this.orderService.getOrdersByOrderedProductId(productId);
+    return orders.map((order) => OrderMapper.toDto(order));
```

Comment:
**suggestion(non-blocking):** you can create a method in the mappers `toDtoList` so it looks more clean